spec/views/profiles/index.html.haml_spec.rb

```diff
@@ -0,0 +1,28 @@
+require "rails_helper"
+
+describe "profiles/index.html.haml" do
+  let!(:profile1) { create(:profile) }
+  let!(:profile2) { create(:profile) }
+
+  before do
+    assign(:profiles, Profile.all)
+  end
+
+  context "when there is at least one profile" do
+    it "shows a list of profiles" do
+      render template: "profiles/index.html.haml"
+
+      expect(rendered).to have_link(profile1.name)
+      expect(rendered).to have_link(profile2.name)
+    end
+  end
+
+  context "when there are no profiles" do
+    it "shows an information about no profiles to list" do
+      Profile.delete_all
```

Comment:
Not the best solution. You create profiles in lines 4 and 5 just to delete them here. We don't need to do this db calls (btw good that you use `delete_all` to reduce db overhead.

I would recommend moving `lets` to `when there is at least one profile context` as an optimal solution here